### PR TITLE
Implemented `guardian-supervisor-strategy` hocon setting

### PR DESCRIPTION
**Pull request by JeffCyr**
This PullRequest makes it possible to change the User Guardian supervision strategy with the guardian-supervisor-strategy` hocon setting.

### CODE REVIEW

**Comment by Silv3rcircl3** [comment]
I think this is a duplicate of #1525 :)

**Comment by JeffCyr** [comment]
Doh!

**Comment by Aaronontheweb** [comment]
@Silv3rcircl3 @JeffCyr yeah, but #1525 looks like it's still a work in progress - need that TestKit.EventFilter to be able to apply to multiple actor systems in order to test this, right?

**Comment by Silv3rcircl3** [comment]
yep, right and theoretically this PR would also need this. I assume that's the reason why @JeffCyr has changed the implementation of these tests compared to the scala version.

**Comment by JeffCyr** [comment]
@Silv3rcircl3 @Aaronontheweb I didn't understand why the scala version used the EventFilter, don't you think it is a bit zealous? The test is more about asserting that the actor has stopped than observing that an exception was thrown from the actor.

File: src/core/Akka.Tests/Actor/ActorSystemSpec.cs [hunk 1]
```diff
@@ -238,6 +239,50 @@ public void Support_using_a_customer_scheduler()
...
+            {
+                actor.Receive<string>((msg, context) => { throw new Exception("Boom"); });
+            });
+
+            var probe = CreateTestProbe(system);
+            probe.Watch(a);
+
+            a.Tell("die");
+
+            var t = probe.ExpectTerminated(a);
```
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Actor/ActorSystemSpec.cs]: Yeah, this looks fine to me - don't think we need #1753 to complete this. The `ExpectTerminated` does the job.

File: src/core/Akka.Tests/Actor/ActorSystemSpec.cs [hunk 2]
```diff
@@ -238,6 +239,50 @@ public void Support_using_a_customer_scheduler()
...
+            var system = ActorSystem.Create(Guid.NewGuid().ToString(), config);
+
+            var a = system.ActorOf(actor =>
+            {
+                actor.Receive<string>((msg, context) => { throw new Exception("Boom"); });
+            });
+
+            a.Tell("die");
+
+            Assert.True(system.WhenTerminated.Wait(1000));
```
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Actor/ActorSystemSpec.cs]: Yep, also looks fine without needing the `EventFilter`

File: src/core/Akka/Actor/ActorRefProvider.cs [hunk 3]
```diff
@@ -155,9 +156,8 @@ public LocalActorRefProvider(string systemName, Settings settings, EventStream e
             _tempNumber = new AtomicCounterLong(1);
             _tempNode = _rootPath / "temp";
 
-            //TODO: _guardianSupervisorStrategyConfigurator = dynamicAccess.createInstanceFor[SupervisorStrategyConfigurator](settings.SupervisorStrategyClass, EmptyImmutableSeq).get
             _systemGuardianStrategy = SupervisorStrategy.DefaultStrategy;
-
+            _userGuardianStrategyConfigurator = SupervisorStrategyConfigurator.CreateConfigurator(Settings.SupervisorStrategyClass);
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/ActorRefProvider.cs]: Looks correct

**Comment by Aaronontheweb** [comment]
Compared to the JVM source and it seemed complete and correct to me.

**Pull request merged**